pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -0,0 +1,259 @@
+        "FREQ:STAR?",
+        "FREQ:STAR %s",
+        "Start frequency in Hz.",
+    )
+
+    freq_stop = Instrument.control(
+        "FREQ:STOP?",
+        "FREQ:STOP %s",
+        "Stop frequency in Hz.",
+    )
+
+    attenuation = Instrument.control(
+        "INP:ATT?",
+        "INP:ATT %s",
+        "Attenuation in dB.",
+    )
+
+    @property
+    def rbw(self):
+        """Resolution bandwidth in Hz. Can be set to 'AUTO'."""
+        return self.values("BAND:RES?")[0]
+
+    @rbw.setter
+    def rbw(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("BAND:RES:AUTO ON")
+        else:
+            self.write(f"BAND:RES {value}")
+
+    @property
+    def vbw(self):
+        """Video bandwidth in Hz. Can be set to 'AUTO'."""
+        return self.values("BAND:VID?")[0]
+
+    @vbw.setter
+    def vbw(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("BAND:VID:AUTO ON")
+        else:
+            self.write(f"BAND:VID {value}")
+
+    # Sweeping ----------------------------------------------------------------
+
+    @property
+    def sweep_time(self):
+        """Sweep time in s. Can be set to 'AUTO'."""
+        return self.values("SWE:TIME?")[0]
+
+    @sweep_time.setter
+    def sweep_time(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("SWE:TIME:AUTO ON")
+        else:
+            self.write(f"SWE:TIME {value}")
+
+    continuous_sweep = Instrument.control(
+        "INIT:CONT?",
+        "INIT:CONT %s",
+        "Continuous (True) or single sweep (False)",
+        validator=strict_discrete_set,
+        values=[True, False],
+        set_process=lambda x: "ON" if x else "OFF",
+        get_process=lambda x: bool(x),
+    )
+
+    def single_sweep(self):
+        """Perform a single sweep with synchronization."""
+        self.write("INIT; *WAI")
+
+    def continue_single_sweep(self):
+        """Continue with single sweep with synchronization."""
+        self.write("INIT:CONM; *WAI")
+
+    # Traces ------------------------------------------------------------------
+
+    def read_trace(self, n_trace=1):
+        """
+        Read trace data.
+
+        :param n_trace: The trace number (1-6). Default is 1.
+        :return: Nnumpy arrays of the trace data (frequency and amplitude).
+        """
+        y = np.array(self.values("TRAC{n_trace}? TRACE{n_trace}"))
+        x = np.linspace(self.freq_start, self.freq_stop, len(y))
+        return x, y
+
+    trace_mode = Instrument.control(
+        "DISP:TRAC:MODE?",
+        "DISP:TRAC:MODE %s",
+        "Trace mode ('WRIT', 'MAXH', 'MINH', 'AVER' or 'VIEW'",
+        validator=strict_discrete_set,
+        values=["WRIT", "MAXH", "MINH", "AVER", "VIEW"],
+    )
+
+    # Markers -----------------------------------------------------------------
+
+    def create_marker(self, num=1, is_delta_marker=False):
```

Comment:
What happens if you create the same marker twice?

Reply:
Nothing special. You get back another Marker instance. Both do work. Do you see an issue with that?

Reply:
In that case the marker switches between being a delta marker and a regular marker every time one of the two `Marker` objects is used (indicated by the label M3 and D3 on the display of the instrument).

Reply:
thanks for clarifying!